### PR TITLE
Make ``get_blocks()`` functions consistently return lists

### PR DESCRIPTION
**Pull request by clinssen**
This is a purely internal refactoring.

Make ``get_equations_blocks()``, ``get_parameters_blocks()`` and other block-getter function on the ``ASTNeuronOrSynapse`` return lists in a consistent manner. Also de-duplicate some code and fix imports.

Motivation for preferring the list option rather than single elements, is because this leaves the way open in the future, where NESTML could have an "import" or "include" mechanism, to make models more modular. Then, two different blocks of the same type (e.g. ``state``, ``parameters``) from two different .nestml files might end up in the same model.

### CODE REVIEW

File: pynestml/utils/ast_utils.py [hunk 1]
```diff
@@ -1878,13 +1902,20 @@ def find_model_by_name(cls, model_name: str, models: Iterable[Union[ASTNeuron, A
         return None
 
     @classmethod
-    def get_convolve_function_calls(cls, ast):
+    def get_convolve_function_calls(cls, nodes: Union[ASTNode, List[ASTNode]]):
         """
         Returns all sum function calls in the handed over meta_model node or one of its children.
         :param ast: a single meta_model node.
```
Comment (pnbabu) [inline pynestml/utils/ast_utils.py:1908]:
```suggestion
        :param nodes: a single or list of AST nodes.
```

File: pynestml/utils/ast_utils.py [hunk 2]
```diff
@@ -1878,13 +1902,20 @@ def find_model_by_name(cls, model_name: str, models: Iterable[Union[ASTNeuron, A
...
         :type ast: ASTNode
```
Comment (pnbabu) [inline pynestml/utils/ast_utils.py:1909]: Can be removed.

File: pynestml/utils/ast_utils.py [hunk 3]
```diff
@@ -1900,7 +1931,7 @@ def contains_convolve_function_call(cls, ast):
     @classmethod
     def get_function_calls(cls, ast_node, function_list):
         """
-        For a handed over list of function names, this method retrieves all functions in the meta_model.
+        For a handed over list of function names, this method retrieves all function calls in the meta_model.
         :param ast_node: a single meta_model node
         :type ast_node: ASTNode
```
Comment (pnbabu) [inline pynestml/utils/ast_utils.py:1936]: Could you remove the type here and add that in the function signature?

File: pynestml/utils/ast_utils.py [hunk 4]
```diff
@@ -1917,3 +1948,11 @@ def get_function_calls(cls, ast_node, function_list):
         vis = ASTHigherOrderVisitor(visit_funcs=fun)
         ast_node.accept(vis)
         return res
+
+    @classmethod
+    def resolve_to_variable_symbol_in_blocks(cls, variable_name: str, blocks: List[ASTBlock]):
```
Comment (pnbabu) [inline pynestml/utils/ast_utils.py:1953]: Could you add a docstring for this function?

**Review by pnbabu — COMMENTED** [review]
Thanks for this clean-up! I have some minor comments.

File: pynestml/codegeneration/nest_code_generator.py [hunk 5]
```diff
@@ -423,7 +425,7 @@ def _get_synapse_model_namespace(self, synapse: ASTSynapse) -> Dict:
             and self.analytic_solver[synapse.get_name()] is not None
         if namespace["uses_analytic_solver"]:
             namespace["analytic_state_variables"] = self.analytic_solver[synapse.get_name()]["state_variables"]
-            namespace["variable_symbols"].update({sym: synapse.get_equations_block().get_scope().resolve_to_symbol(
+            namespace["variable_symbols"].update({sym: synapse.get_equations_blocks()[0].get_scope().resolve_to_symbol(
```
[the remark this answers is not in the document]
Comment (clinssen) [reply pynestml/codegeneration/nest_code_generator.py:428]: I prefer just passing the synapse object as the function parameter and grabbing it again from there.

File: pynestml/codegeneration/nest_code_generator.py [hunk 6]
```diff
@@ -64,9 +63,7 @@
 def find_spiking_post_port(synapse, namespace):
```
[the remark this answers is not in the document]
Comment (clinssen) [reply pynestml/codegeneration/nest_code_generator.py:63]: Yes! But this is already being refactored in one of the other PRs, so I'd like to keep this separate for now.

File: pynestml/utils/ast_utils.py (hunk 2 above)
Comment (clinssen) [reply pynestml/utils/ast_utils.py:1909]: ?

Comment (pnbabu) [reply pynestml/utils/ast_utils.py]: I meant the type in docstring can be removed.

**Review by pnbabu — APPROVED** [review]
Looks good to me!

File: pynestml/utils/ast_utils.py (hunk 2 above)
Comment (clinssen) [reply pynestml/utils/ast_utils.py]: Oh, of course, sorry.

**Comment by clinssen** [comment]
Cheers!

**Pull request merged**